File: datasets/C++/Chatgpt/Algorithm/easy1.cpp
#include <iostream>
#include <set>
#include <string>
using namespace std;

bool isValid(const string &s) {
    for (size_t i = 1; i < s.size(); ++i) {
        if (s[i] == s[i - 1])
            return false;
    }
    return true;
}

int alternate(string s) {
    set<char> chars(s.begin(), s.end());
    int maxLen = 0;

    for (auto it1 = chars.begin(); it1 != chars.end(); ++it1) {
        auto it2 = it1;
        ++it2;
        for (; it2 != chars.end(); ++it2) {
            string temp = "";
            for (char c : s) {
                if (c == *it1 || c == *it2)
                    temp += c;
            }
            if (isValid(temp))
                maxLen = max(maxLen, (int)temp.length());
        }
    }
    return maxLen;
}

int main() {
    int n;
    string s;
    cout << "Enter string length: ";
    cin >> n;
    cout << "Enter the string: ";
    cin >> s;

    cout << "Longest alternating string length: " << alternate(s) << endl;
    return 0;
}

File: datasets/Python/Chatgpt/Algorithm/easy1.py
def alternate(s):
    unique_chars = list(set(s))
    max_length = 0

    for i in range(len(unique_chars)):
        for j in range(i + 1, len(unique_chars)):
            ch1 = unique_chars[i]
            ch2 = unique_chars[j]
            filtered = [c for c in s if c == ch1 or c == ch2]

            if all(filtered[k] != filtered[k+1] for k in range(len(filtered)-1)):
                max_length = max(max_length, len(filtered))
    return max_length

# Manual input
n = int(input("Enter string length: "))
s = input("Enter the string: ").strip()
print("Longest alternating string length:", alternate(s))